library/python-library/src/main/java/uk/gov/gchq/gaffer/python/operation/SendAndGetDataFromContainer.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2019 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.python.operation;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.net.Socket;
+import java.util.concurrent.TimeUnit;
+
+public class SendAndGetDataFromContainer {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SendAndGetDataFromContainer.class);
+
+    public SendAndGetDataFromContainer() {
+    }
+
+    /**
+     * @param operation the RunPythonScript operation
+     * @param port the port of the docker client where the data will be passed
+     * @return Sets up and closes container
+     * @throws InterruptedException should this fail, this will be thrown
+     * @throws IOException this will be thrown if non-compliant data is sent
+     */
+    public StringBuilder setUpAndCloseContainer(final RunPythonScript operation, final String port) throws InterruptedException, IOException {
+        // Keep trying to connect to container and give the container some time to load up
+        boolean failedToConnect = true;
+        IOException error = null;
+        Socket clientSocket = null;
+        DataInputStream in = null;
+        Thread.sleep(1000);
+        LOGGER.info("Attempting to connect with the container...");
+        for (int i = 0; i < 100; i++) {
+            try {
+                clientSocket = new Socket("127.0.0.1", Integer.parseInt(port));
```

Comment:
Do you need to keep reassigning this? If not, could the assignment be moved out of the loop and into a try-with-resources? Currently there is a resource leak if (failedToConnect)